Mathlib/Topology/Separation.lean

```diff
@@ -94,6 +101,12 @@ If the space is also compact:
   is equivalent to being a `TotallySeparatedSpace`.
 * `ConnectedComponents.t2`: `ConnectedComponents X` is T₂ for `X` T₂ and compact.
 
+### Regular spaces
+
+If the space is also Lindelöf:
```

Comment:
What I meant by "references" was to add external references, like a wikipedia entry or (better) a book/chapter of a book.

Reply:
Well, https://en.wikipedia.org/wiki/Separation_axiom is already referenced. I'll add https://en.wikipedia.org/wiki/Normal_space since it asserts the specific result. I could add a book, but this is pretty standard stuff so honestly any graduate-level topology textbook (Engelking, Munkres, Willard, etc) would have this result.

I have Engelking handy and this is a consequence of the first few results in 5.1 Paracompact spaces (but that's not a great reference as Engelking loves to overload separation axioms into his definitions for covering properties, hiding the regularity requirement).

Reply:
My only point is that I find it nice that for "classical" results we indicate that what is being formalized can be also read on paper, to make a nice connection with the "real world". This should really not be a pain for you, I was just suggesting this as you are clearly much more knowledgeable on the existing grad/undergrad literature.

Reply:
This does reveal that eventually this should be improved to show that every regular Lindelof space is fully normal (equivalently in this context, paracompact). But TBH this was a specific TODO that I was trying to address at https://github.com/leanprover-community/mathlib4/blob/04c29e2530bb437a841451567fa770f65183f6b5/Mathlib/Topology/Separation.lean#L2291 so I didn't want to go overboard.

Reply:
I'll head to the office tomorrow and see if there's a clean reference that uses modern definitions of separation axioms and covering properties. (I think Willard does this carefully.)

Reply:
addressed in 53b8f7b